### PR TITLE
Copy authority rather than retain to avoid possible crash

### PR DESCRIPTION
**Pull request by jasoncoolmax**

### CODE REVIEW

File: ADAL/src/cache/ADTokenCacheAccessor.m [hunk 1]
```diff
@@ -55,8 +55,7 @@ - (id)initWithDataSource:(id<ADTokenCacheDataSource>)dataSource
     _dataSource = dataSource;
     SAFE_ARC_RETAIN(dataSource);
     
-    _authority = authority;
-    SAFE_ARC_RETAIN(authority);
+    _authority = [authority copy];
```
Comment (jasoncoolmax) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: the known crash happened at line 75 if this file

Comment (RPangrle) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: This is too late. I meant to make sure in ADAuthenticationContext that we're not just retaining the passed in input, which we aren't. Doing a copy here is unnecessary.

Comment (jasoncoolmax) [inline ADAL/src/cache/ADTokenCacheAccessor.m]: Make sense :)

**Comment by jasoncoolmax** [comment]
not needed

**Pull request closed without merging**